keanu-project/src/main/java/io/improbable/keanu/plating/loop/Loop.java

```diff
@@ -0,0 +1,137 @@
+    public static final int DEFAULT_MAX_COUNT = 100;
+    private final Plates plates;
+    private final boolean throwWhenMaxCountIsReached;
+
+    /**
+     * package-private because it is intended to be created by the LoopBuilder
+     *
+     * @param plates
+     * @param throwWhenMaxCountIsReached
+     */
+    Loop(Plates plates, boolean throwWhenMaxCountIsReached) {
+        this.plates = plates;
+        this.throwWhenMaxCountIsReached = throwWhenMaxCountIsReached;
+    }
+
+    public Plates getPlates() {
+        return plates;
+    }
+
+    /**
+     * A factory method for creating a loop
+     * It automatically labels the initial state correctly for you.
+     *
+     * @param initialState - a single Vertex that defines the loop's base case.
+     * @param <V>          the input type
+     * @return a builder object
+     */
+    public static <V extends Vertex<?>> LoopBuilder startingFrom(V initialState) {
+        if (initialState.getLabel() == null) {
+            initialState.setLabel(VALUE_OUT_LABEL);
+        }
+        return startingFrom(ImmutableList.of(initialState));
+    }
+
+    /**
+     * A factory method for creating a loop
+     *
+     * @param first  - the first Vertex (mandatory)
+     * @param others - other Vertices (optional)
+     * @param <V>    the input type
+     * @return a builder object
+     */
+    public static <V extends Vertex<?>> LoopBuilder startingFrom(V first, V... others) {
+        return startingFrom(ImmutableList.<V>builder().add(first).add(others).build());
+    }
+
+    /**
+     * A factory method for creating a loop
+     *
+     * @param initialState - the collection of vertices that define the loop's base case
+     * @param <V>          the input type
+     * @return a builder object
+     */
+    public static <V extends Vertex<?>> LoopBuilder startingFrom(Collection<V> initialState) {
+        return new LoopBuilder(initialState);
+    }
+
+    /**
+     * @param <V> the output type
+     * @return the output of the Loop (i.e. the output Vertex from the final Plate)
+     * @throws LoopException if the loop was too short and hit its maximum unrolled size
+     */
+    public <V extends Vertex<? extends Tensor<?>>> V getOutput() throws LoopException {
+        Plate finalPlate = plates.asList().get(plates.size() - 1);
+        checkIfMaxCountHasBeenReached(finalPlate);
+        return finalPlate.get(VALUE_OUT_LABEL);
+    }
+
+    private void checkIfMaxCountHasBeenReached(Plate plate) throws LoopException {
+        BoolVertex stillLooping = plate.get(STILL_LOOPING_LABEL);
+        if (Arrays.stream(stillLooping.getValue().asFlatArray()).anyMatch(v -> v == true)) {
+            String errorMessage = "Loop has exceeded its max count " + plates.size();
+            if (throwWhenMaxCountIsReached) {
+                throw new LoopException(errorMessage);
+            } else {
+                log.warn(errorMessage);
```

Comment:
Why WARN level? Maybe INFO since the user explicitly told us it's ok?

Reply:
fair enough - done